features/step_definitions/plans/application_plans_steps.rb

```diff
@@ -131,3 +131,27 @@
   assert_content "The plan was deleted", wait: 10
   step(%(I should not see plan "#{name}"))
 end
+
+And "an application plan that is not default" do
+  @plan = FactoryBot.create(:application_plan, name: "Magic", issuer: @provider.default_service)
+  assert_not @plan.master?
+end
+
+Given "the application plan is hidden" do
+  assert @plan.hidden?
```

Comment:
Looking at this step, what is the difference between `Marking a Hidden plan as default` and `Marking plan as default` scenarios?

Reply:
Yes, from a code point of view they are redundant. There's a small difference though: this scenario makes sure the plan is hidden, that is the condition. In the other scenario the plan state is _by default_, whatever it is. We could also change it to "Marking a published plan as default". I honestly thought plans were created published by default.

Reply:
yeah, it makes some sense that we test making a public plan as published, otherwise it is just repeated test. By default plan is hidden probably to make sure plan is finalized before it becomes available to users.

Reply:
Thank you for your help @akostadinov, I've done what you suggested and added a new step assuring we are working with a published plan.